L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleGen.cc

```diff
@@ -40,53 +205,245 @@ HGCalTriggerNtupleGen(const edm::ParameterSet& conf):HGCalTriggerNtupleBase(conf
+
+      unsigned nlayers = 40;
+      if (myTrack.noEndVertex())  // || myTrack.genpartIndex()>=0)
+      {
+        HGCal_helpers::coordinates propcoords;
+        bool reachesHGCal = toHGCalPropagator.propagate(
+            myTrack.momentum(), myTrack.vertex().position(), myTrack.charge(), propcoords);
+        vtx = propcoords.toVector();
+
+        if (reachesHGCal && vtx.Rho() < 160 && vtx.Rho() > 25) {
+          reachedEE = 2;
+          double dpt = 0;
+
+          for (int i = 0; i < myTrack.nDaughters(); ++i) dpt += myTrack.daughter(i).momentum().pt();
+          if (abs(myTrack.type()) == 11) fbrem = dpt / myTrack.momentum().pt();
+        } else if (reachesHGCal && vtx.Rho() > 160)
+          reachedEE = 1;
+
+        HGCal_helpers::simpleTrackPropagator indiv_particleProp(aField_);
+        for (unsigned il = 0; il < nlayers; ++il) {
+          const float charge = myTrack.charge();
+          indiv_particleProp.setPropagationTargetZ(layerPositions_[il]);
+          HGCal_helpers::coordinates propCoords;
+          indiv_particleProp.propagate(myTrack.momentum(), myTrack.vertex().position(), charge,
+                                       propCoords);
+
+          xp.push_back(propCoords.x);
+          yp.push_back(propCoords.y);
+          zp.push_back(propCoords.z);
+        }
+      } else {
+        vtx = myTrack.endVertex().position();
+      }
+      auto orig_vtx = myTrack.vertex().position();
+
+      allselectedgentracks.push_back(&mySimEvent_->track(i));
+      // fill branches
+      genpart_eta_.push_back(myTrack.momentum().eta());
+      genpart_phi_.push_back(myTrack.momentum().phi());
+      genpart_pt_.push_back(myTrack.momentum().pt());
+      genpart_energy_.push_back(myTrack.momentum().energy());
+      genpart_dvx_.push_back(vtx.x());
+      genpart_dvy_.push_back(vtx.y());
+      genpart_dvz_.push_back(vtx.z());
+
+      genpart_ovx_.push_back(orig_vtx.x());
+      genpart_ovy_.push_back(orig_vtx.y());
+      genpart_ovz_.push_back(orig_vtx.z());
+
+      HGCal_helpers::coordinates hitsHGCal;
+      toHGCalPropagator.propagate(myTrack.momentum(), orig_vtx, myTrack.charge(), hitsHGCal);
+
+      genpart_exphi_.push_back(hitsHGCal.phi);
+      genpart_exeta_.push_back(hitsHGCal.eta);
+      genpart_exx_.push_back(hitsHGCal.x);
+      genpart_exy_.push_back(hitsHGCal.y);
+
+      genpart_fbrem_.push_back(fbrem);
+      genpart_pid_.push_back(myTrack.type());
+      genpart_gen_.push_back(myTrack.genpartIndex());
+      genpart_reachedEE_.push_back(reachedEE);
+      genpart_fromBeamPipe_.push_back(true);
+
+      genpart_posx_.push_back(xp);
+      genpart_posy_.push_back(yp);
+      genpart_posz_.push_back(zp);
+    }
+
+
+    edm::Handle<std::vector<reco::GenParticle>> genParticlesHandle;
+    iEvent.getByToken(gen_token_, genParticlesHandle);
+    gen_n_ = genParticlesHandle->size();
+
+    for (std::vector<reco::GenParticle>::const_iterator it_p = genParticlesHandle->begin();
```

Comment:
Would be more readable to use a range-based loop

Reply:
matter of taste I guess...

Reply:
ok